deployer/src/deployment/queue.rs

```diff
@@ -432,18 +409,27 @@ async fn run_pre_deploy_tests(
         .arg("--color=always")
         .current_dir(project_path)
         .stdout(Stdio::piped())
-        .spawn()
-        .map_err(TestError::Run)?;
-
-    let stdout = cmd.stdout.take().unwrap();
-    let stdout_reader = BufReader::new(stdout);
-    for line in stdout_reader.lines().flatten() {
-        if let Err(error) = write.send(format!("{}\n", line.trim_end_matches('\n'))) {
-            error!("failed to send to pipe: {error}");
+        .stderr(Stdio::piped());
+
+    let mut handle = cmd.spawn().map_err(TestError::Run)?;
+    let tx2 = tx.clone();
+    let reader = tokio::io::BufReader::new(handle.stdout.take().unwrap());
+    tokio::spawn(async move {
+        let mut lines = reader.lines();
+        while let Some(line) = lines.next_line().await.unwrap() {
+            let _ = tx.send(line).await.map_err(|e| error!("{e}"));
         }
-    }
+    });
+    let reader = tokio::io::BufReader::new(handle.stderr.take().unwrap());
+    tokio::spawn(async move {
+        let mut lines = reader.lines();
+        while let Some(line) = lines.next_line().await.unwrap() {
+            let _ = tx2.send(line).await.map_err(|e| error!("{e}"));
+        }
+    });
+    let status = handle.wait().await.map_err(TestError::Run)?;
```

Comment:
Is opting not to use os_pipe here relevant to the bug fix (missing crates download output)? It was a pretty good and simple way of combining stderr and stdout, what is the advantage of doing it this way?

Reply:
Oh, nevermind, it seems we didn't actually use os_pipe here, and we ignored stderr. So this change should be an improvement. We might still consider using os_pipe to combine stderr and stdout, WDYT?

Reply:
Main thing is that os_pipe and pipe and crossbeam are blocking, so I changed them all out for tokio async readers, which solved the bug, and should be more sound in an async app.